chain/client/src/info.rs

```diff
@@ -877,23 +917,23 @@ impl ValidatorProductionStats {
             num_expected_chunks_per_shard: info.num_expected_chunks_per_shard,
             num_produced_endorsements_per_shard: info.num_produced_endorsements_per_shard,
             num_expected_endorsements_per_shard: info.num_expected_endorsements_per_shard,
-        }
+        })
     }
 }
 
-/// Converts EpochValidatorInfo into a vector of ValidatorProductionStats.
-fn get_validator_epoch_stats(
+/// Converts EpochValidatorInfo into a vector of ValidatorProductionStatus.
+fn get_validator_production_status(
     current_validator_epoch_info: EpochValidatorInfo,
-) -> Vec<ValidatorProductionStats> {
-    let mut stats = vec![];
+) -> Vec<ValidatorProductionStatus> {
+    let mut status = vec![];
     // Record kickouts to replace latest stats of kicked out validators with zeros.
     for kickout in current_validator_epoch_info.prev_epoch_kickout {
```

Comment:
sanity check - Should that be prev_epoch or prev_prev_epoch kickouts? Does this work as following?
* Validator is bad in epoch T
* It is removed from the validator list of epoch T + 2 
* Is its status kickout in epoch T + 1 ? It's not on chain right? Is this info derived from the validator sets?

Reply:
Good point. See this code: https://github.com/near/nearcore/blob/3bdc50d2a1c537bd300ce0c0df2813cd574c3ee9/chain/epoch-manager/src/lib.rs#L1563

So the prev_epoch_kickout (of T+1) is computed from the next epoch info (T+2), and the kickouts are referring to the validator stats from T.

Reply:
An alternative approach would be to forget about kickouts and just sweep all the currently set metrics and unset them and then just set the metrics for current validators. Not sure about perf so maybe do it only at the first block of epoch.